src/lib/components/wizard.component.ts

```diff
@@ -107,24 +107,81 @@ export class WizardComponent implements OnChanges, AfterContentInit {
 
   /**
    * The initially selected step, represented by its index
+   * Beware: This initial default is only used if no wizard step has been enhanced with the `selected` directive
    */
   @Input()
-  public defaultStepIndex = 0;
+  public get defaultStepIndex(): number {
+    // This value can be either:
+    // - the index of a wizard step with a `selected` directive, or
+    // - the default step index, set in the [[WizardComponent]]
+
+    const foundDefaultStep = this.wizardSteps.find(step => step.defaultSelected);
+
+    if (foundDefaultStep) {
+      return this.getIndexOfStep(foundDefaultStep);
+    } else {
+      return this._defaultStepIndex;
+    }
+  }
+  public set defaultStepIndex(defaultStepIndex: number) {
+    this._defaultStepIndex = defaultStepIndex;
+  }
+  private _defaultStepIndex = 0;
+
+  /**
+   * The WizardStep object belonging to the currently visible and selected step.
+   * The currentStep is always the currently selected wizard step.
+   * The currentStep can be either completed, if it was visited earlier,
+   * or not completed, if it is visited for the first time or its state is currently out of date.
+   *
+   * If this wizard contains no steps, currentStep is null
+   */
+  public get currentStep(): WizardStep {
+    if (this.hasStep(this.currentStepIndex)) {
+      return this.wizardSteps[this.currentStepIndex];
+    } else {
+      return null;
+    }
+  }
 
   /**
    * True, if the navigation bar shouldn't be used for navigating
    */
   @Input()
   public disableNavigationBar = false;
 
+  /**
+   * An array representation of all wizard steps belonging to this model
+   */
+  public get wizardSteps(): WizardStep[] {
```

Comment:
Do we really require three different names to interact with the wizard steps? (`wizardStepsQueryList`, `wizardSteps` and `_wizardSteps`). Can we reduce this to only `wizardStepsQueryList` and `wizardSteps`? In my opinion this would make the code more easy to understand

Reply:
Yes, it is possible to remove `_wizardSteps` at the cost of calling `wizardStepsQueryList.toArray()` each time. It may slow things down just a tiny bit, because internally `QueryList.toArray()` [calls](https://github.com/angular/angular/blob/91c7b451d5323f8755b539dc8293deffa9d71cc7/packages/core/src/linker/query_list.ts#L100) `slice()` to produce a copy of array instance, and `wizardSteps` is accessed pretty often. I am not sure what is more important, improved readability or the negative impact on performance?

```diff
diff --git a/src/lib/components/wizard.component.ts b/src/lib/components/wizard.component.ts
index d0a1870..2bee646 100644
--- a/src/lib/components/wizard.component.ts
+++ b/src/lib/components/wizard.component.ts
@@ -146,9 +146,8 @@ export class WizardComponent implements AfterContentInit, OnChanges {
    * An array representation of all wizard steps belonging to this model
    */
   public get wizardSteps(): WizardStep[] {
-    return this._wizardSteps;
+    return this.wizardStepsQueryList.toArray();
   }
-  private _wizardSteps: WizardStep[] = [];
 
   /**
    * The index of the currently visible and selected step inside the wizardSteps QueryList.
@@ -252,17 +251,15 @@ export class WizardComponent implements AfterContentInit, OnChanges {
   }
 
   /**
-   * Updates the wizard steps to the new array
+   * Handle wizard steps update
    *
    * @param wizardSteps The updated wizard steps
    */
   private updateWizardSteps(wizardSteps: WizardStep[]): void {
-    // the wizard is currently not in the initialization phase
+    // Adjust currentStepIndex if necessary (if the wizard is not stillin the initialization phase)
     if (this.wizardSteps.length > 0 && this.currentStepIndex > -1) {
       this.currentStepIndex = wizardSteps.indexOf(this.wizardSteps[this.currentStepIndex]);
     }
-
-    this._wizardSteps = wizardSteps;
   }
 
   /**
```

Reply:
I think we should prioritize performance in this case